src/shogun/labels/BinaryLabelEncoder.h

```diff
@@ -0,0 +1,117 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ *
+ */
+#ifndef _BINARYLABELENCODER__H__
+#define _BINARYLABELENCODER__H__
+
+#include <memory>
+#include <shogun/base/SGObject.h>
+#include <shogun/labels/BinaryLabels.h>
+#include <shogun/labels/DenseLabels.h>
+#include <shogun/labels/LabelEncoder.h>
+#include <shogun/lib/SGVector.h>
+
+namespace shogun
+{
+
+	class BinaryLabelEncoder : public LabelEncoder
+	{
+	public:
+		BinaryLabelEncoder() = default;
+
+		~BinaryLabelEncoder() = default;
+
+		/** Fit label encoder
+		 *
+		 * @param Target values.
+		 * @return SGVector which contains unique labels.
+		 */
+		SGVector<float64_t> fit(const std::shared_ptr<Labels>& labs) override
+		{
+			const auto result_vector = labs->as<DenseLabels>()->get_labels();
+			fit_impl(result_vector);
```

Comment:
I don't think it will work having a single `fit_impl`. I think BinaryLabels require {-1,+1}, but I could have labels {0,10}. So I would expect that mapping to happen in the BinaryLabelEncoder? Otherwise I will just get {0,10} here, right?

Reply:
ah that is done in transform, but then how do you keep the original mapping? What if I fit to a set that starts with label 0 and then test set first label is 10? Would I still get the same mapping?

Reply:
i assume that the labels of fit should have the same categories as the labels of transform.

Reply:
Yes, that makes sense I think. I was just a bit confused at the start :D